lib/matplotlib/tests/test_backend_qt.py

```diff
@@ -47,8 +65,48 @@ def test_fig_close(backend):
     assert init_figs == Gcf.figs
 
 
+@pytest.mark.backend('Qt5Agg')
+@pytest.mark.parametrize("target, kwargs, windows_left", [
+    (plt.show, {"block": True}, 0),
+    (plt.pause, {"interval": 10}, 1)
+])
+def test_sigint(qt_core, platform_simulate_ctrl_c, target,
+                kwargs, windows_left):
+    plt.ion()
+    fig, ax = plt.subplots()
+    xdata, ydata = [], []
+    ln, = plt.plot([], [], 'ro')
+
+    def init():
+        ax.set_xlim(0, 100)
+        ax.set_ylim(0, 100)
+        return ln,
+
+    def update(frame):
+        if frame < 50:
+            xdata.append(frame)
+            ydata.append(frame)
+            ln.set_data(xdata, ydata)
+            return ln,
+        elif frame == 50:
+            platform_simulate_ctrl_c()
+            return ln,
+        else:
+            return ln,
+
+    ani = FuncAnimation(fig, update, frames=range(0, 100),
+                        init_func=init, blit=True, interval=5)
+    try:
+        target(**kwargs)
+    except KeyboardInterrupt as e:
+        assert len(Gcf.get_all_fig_managers()) == windows_left
+        return
+    assert False
```

Comment:
The test won't pass with this.

Reply:
With line 102 and no pyplot.close() in the handler which we can't have there since we can't import pyplot in backend_qt5, yes, it won't. Otherwise its OK?

Reply:
Oh, I missed the early return; please add an explicit message on the assertion then.